### PR TITLE
feat(codex-ui): tab component

### PR DESCRIPTION
**Pull request by elizachi**
Now we have tab component in the playground
image

### CODE REVIEW

File: codex-ui/dev/Playground.vue [hunk 1]
```diff
@@ -204,6 +204,23 @@
...
+    <Heading :level="3"> Tab </Heading>
+    <Tab :name="'Home'" />
+    <br />
+    <Tab
+      :name="'Home'"
+      :src="'../static/example-avatar.png'"
+    />
+    <br />
+    <Tab
+      :name="'Home'"
```
Comment (TatianaFomina) [inline codex-ui/dev/Playground.vue:216]:
```suggestion
      name="Home"
```

File: codex-ui/dev/Playground.vue [hunk 2]
```diff
@@ -204,6 +204,23 @@
...
+      :icon="'Plus'"
+    />
+    <br />
+    <Tab
+      :name="'Home'"
+      :show-cross="true"
```
Comment (TatianaFomina) [inline codex-ui/dev/Playground.vue:222]: maybe use prop `closable` instead of `show-cross`?

File: codex-ui/src/vue/tabbar/Tab.types.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/**
+ * The state of tab
+ */
+export type TabState = 'default' | 'current';
```
Comment (TatianaFomina) [inline codex-ui/src/vue/tabbar/Tab.types.ts:4]: Maybe `'active'` instead of `'current'`?

File: codex-ui/src/vue/tabbar/Tab.vue [hunk 4]
```diff
@@ -0,0 +1,117 @@
+<template>
+  <div :class="[$style.tab]">
+    <div v-if="src">
+      <img
+        :src="src"
+        :class="[$style['tab__left__image']]"
```
Comment (TatianaFomina) [inline codex-ui/src/vue/tabbar/Tab.vue:6]: wrong classname. `__` is used to split elements, there can only be one level of nesting within a block. You need to get rid of second `__` here and further

Sooo, I suggest to rename to 'tab__image'

File: codex-ui/src/vue/tabbar/Tab.vue [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+  defineProps<{
+    /**
+     * Name of the tab item
+     */
+    name: string;
+
+    /**
+     * If true we have cross icon on the right
+     */
+    closable?: boolean;
```
Comment (e11sy) [inline codex-ui/src/vue/tabbar/Tab.vue:44]: maybe find better naming?

File: codex-ui/src/vue/tabbar/Tab.vue [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+    /**
+     * Current tab state
+     */
+    state: TabState;
```
Comment (TatianaFomina) [inline codex-ui/src/vue/tabbar/Tab.vue:49]: if there can only be 2 states, maybe it is easier to have just `isActive?: boolean` prop?

File: codex-ui/src/vue/tabbar/Tab.vue [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+    /**
+     * If we have image in the left slot
+     */
+    picture?: string;
```
Comment (TatianaFomina) [inline codex-ui/src/vue/tabbar/Tab.vue:54]: is it picture url? comment should contain this info

File: codex-ui/src/vue/tabbar/Tab.vue [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    --bg: var(--base--bg-secondary-hover);
+    --color: var(--base--text-secondary);
+  }
+
+  &__current {
+    --bg: var(--base--bg-secondary-hover);
+    --color: var(--base--text-solid-foreground);
+  }
+
+  padding: var(--v-padding) var(--h-padding) var(--v-padding) var(--h-padding);
```
Comment (TatianaFomina) [inline codex-ui/src/vue/tabbar/Tab.vue:98]: the same as
```suggestion
  padding: var(--v-padding) var(--h-padding);
```

File: codex-ui/src/vue/tabbar/Tab.vue (hunk 5 above)
Comment (elizachi) [reply codex-ui/src/vue/tabbar/Tab.vue:44]: Before renaming we had "showCross" property. Do you have any ideas?

File: codex-ui/src/vue/components/tabbar/Tab.types.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
+/**
+ * The state of tab
+ */
+export type IsActive = true | false;
```
Comment (TatianaFomina) [inline codex-ui/src/vue/components/tabbar/Tab.types.ts:4]: this is the same as `boolean`

File: codex-ui/src/vue/components/tabbar/Tab.vue [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+<script setup lang="ts">
+import { IsActive } from './Tab.types';
+import Icon from '../icon/Icon.vue';
+
+withDefaults(
+  defineProps<{
+    /**
+     * Name of the tab item
+     */
+    name: string;
```
Comment (TatianaFomina) [inline codex-ui/src/vue/components/tabbar/Tab.vue:39]: maybe `title` or `text`?

**Review by TatianaFomina — APPROVED** [review]

File: codex-ui/src/vue/components/tabbar/Tab.types.ts (hunk 9 above)
Comment (neSpecc) [inline codex-ui/src/vue/components/tabbar/Tab.types.ts]: strange name. maybe `TabActiveState` ?

**Review by neSpecc — APPROVED** [review]

File: codex-ui/src/vue/components/tabbar/Tab.vue [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+    /**
+     * If true we have cross icon on the right
+     */
+    closable?: boolean;
+
+    /**
+     * Current tab state
+     */
+    isActive: TabActiveState;
```
Comment (TatianaFomina) [inline codex-ui/src/vue/components/tabbar/Tab.vue:57]: You can just specify boolean here

**Pull request merged**